_posts/2021-12-31-trino-2021-a-year-of-growth.md

```diff
@@ -0,0 +1,426 @@
+* [contains](https://trino.io/docs/current/functions/ipaddress.html#ip-address-contains) 
+function for checking if a CIDR contains an IP address.
+* [`listagg`]({{site.url}}/docs/current/functions/aggregate.html#listagg)
+function returns concatenated values seperated by a specified separator.
+* [soundex](https://trino.io/docs/current/functions/string.html#soundex) function
+that checks phonetic similarity of two strings.
+* [format_number](https://trino.io/docs/current/functions/conversion.html#format_number) function.
+* [`SET TIME ZONE`]({{site.url}}/docs/current/sql/set-time-zone.html) to set the
+ current time zone for the session.
+* Arbitrary queries in [`SHOW STATS`]({{site.url}}/docs/current/sql/show-stats.html).
+* `CURRENT_CATALOG` and `CURRENT_SCHEMA` session functions. 
+* `TRUNCATE TABLE` which allows for a more efficient delete.
+* `DENY` statement, which enables you to remove a user or groups access via SQL.
+* `IN <catalog>` clause to `CREATE ROLE`, `DROP ROLE`, `GRANT ROLE`, 
+`REVOKE ROLE`, and `SET ROLE` to specify the target catalog of the statement 
+instead of using the current session catalog.
+
+### Query processing improvements
+
+* Added support for automatic query retries (this feature is very experimental
+with some limitations for now).
+* Transparent query retries.
+* Updated the behavior of `ROW` to `JSON` cast to produce `JSON` objects instead
+of `JSON` arrays. 
+* Column and table lineage tracking in `QueryCompletedEvent`.
+
+## Performance improvements
+
+Improved performance for the following operations:
+
+* Querying Parquet data for files containing column indexes.
+* Reading dictionary-encoded Parquet files.
+* Queries using [`rank()`]({{site.url}}/docs/current/functions/window.html#rank) window function.
+* Queries using [`sum()`]({{site.url}}/docs/current/functions/aggregate.html#sum)
+and [`avg()`]({{site.url}}/docs/current/functions/aggregate.html#avg) for 
+decimal types.
+* Queries using `GROUP BY` with single grouping column.
+* Aggregation on decimal values.
+* Evaluation of the WHERE and SELECT clause.
+* Computing the product of decimal values with precision larger than 19.
+* Queries that process row or array data.
+* Queries that contain a `DISTINCT` clause.
+* Reduced memory usage and improved performance of joins.
+* `ORDER BY LIMIT` performance was improved when data was pre-sorted.
+* Node-local Dynamic Filtering
+
+## Security
+
+Added the following improvements and features relevant for authentication, 
+authorization and integration with other security systems:
+
+* Automatic configuration of TLS for 
+[secure internal communication]({{site.url}}/docs/current/security/internal-communication.html).
+This removes the need to provision per-worker TLS certificates.
+* Access control for materialized views.
+* OAuth2/OIDC opaque access tokens.
+* Configuring multiple password authentication plugins.
```

Comment:
Link to https://trino.io/docs/current/security/authentication-types.html#multiple-password-authenticators